### ISSUE TITLE
"System.ArgumentException: Parameter is not valid"

### ISSUE THREAD

**Issue by SpaceManiac** [tgstation/IconDiffBot#17]
https://github.com/tgstation/tgstation/runs/1408540

```
System.ArgumentException: Parameter is not valid.
   at System.Drawing.Image.get_Width()
   at IconDiffBot.Core.DiffGenerator.ExtractImages(Dmi dmi, Stream stream) in /src/IconDiffBot/DiffGenerator.cs:line 222
   at IconDiffBot.Core.DiffGenerator.GenerateDiffs(Stream before, Stream after) in /src/IconDiffBot/DiffGenerator.cs:line 359
   at IconDiffBot.Core.PayloadProcessor.<>c__DisplayClass10_1.<<GenerateDiffs>g__DiffDmi|0>d.MoveNext() in /src/IconDiffBot/PayloadProcessor.cs:line 232
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at IconDiffBot.Core.PayloadProcessor.<GenerateDiffs>d__10.MoveNext() in /src/IconDiffBot/PayloadProcessor.cs:line 266
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at IconDiffBot.Core.PayloadProcessor.<ScanPullRequestImpl>d__9.MoveNext() in /src/IconDiffBot/PayloadProcessor.cs:line 153
```

**Comment by Cyberboss** [comment]
This happens with certain .dmi's but only on the Linux builds (probably because System.Drawing is in beta there)

**Comment by Cyberboss** [comment]
`libpng error: invalid after png_start_read_image or png_read_update_info` uhh...

**Comment by Cyberboss** [comment]
This actually isn't limited to .NET. Mono's implementation does the same thing. I'd like to know what makes this .dmi different from the others
```
libpng error: invalid after png_start_read_image or png_read_update_info

Unhandled Exception: System.ArgumentException: A null reference or invalid value was found [GDI+ status: InvalidParameter]
   at System.Drawing.GDIPlus.CheckStatus(Status status)
   at System.Drawing.Image.get_Width()
   at IconDiffBot.Core.DiffGenerator.ExtractImages(Dmi dmi, Stream stream) in /src/IconDiffBot/DiffGenerator.cs:line 222
   at IconDiffBot.Core.DiffGenerator.GenerateDiffs(Stream before, Stream after) in /src/IconDiffBot/DiffGenerator.cs:line 359
   at IconDiffBot.Program.<Main>d__1.MoveNext() in /src/IconDiffBot/Program.cs:line 26
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at IconDiffBot.Program.<Main>(String[] args)
```
Again, only on linux

**Comment by Cyberboss** [comment]
@lzimann I don't suppose this is familiar to you?

**Comment by lzimann** [comment]
nope

**Comment by SpaceManiac** [comment]
The image is paletted (2-bit colormap) rather than 8-bit RGBA, maybe that has something to do with it?

**Comment by lzimann** [comment]
if that's the issue then it has happened to me, i just force RGBA on it [everytime](https://github.com/lzimann/IconDiffBot/blob/master/icons.py#L77)

**Comment by Arokha** [comment]
https://github.com/VOREStation/VOREStation/blob/f0f03c1d6dbc9e759c7dd9aac8d76c78acc02d65/icons/mob/ties_vr.dmi

This .dmi has this issue.

**Comment by Cyberboss** [comment]
Fixed via #25 #27 #28 and #29
